documentation/blog.md

```diff
@@ -0,0 +1,27 @@
+As you already know, YaST was converted from YCP to Ruby some time ago. At least we repeat it quite often ;-) However, this conversion was done on language basis. Some old design decissions and principles stayed. One of them is usage of SCR for accessing underlying configuration files.
+
+SCR was designed together with YaST. It uses concept of "agents" for accessing configuration files. These agents contains a description of configuration file using own format. Moreover SCR offers location transparency. You can e.g. work with a file in system or in a chrooted environment. However, this piece of code is proprietaty and limited by different quality of agents. It is also written in C++ and development is done only in SUSE. From this and some other reasons we decided to replace proprietary SCR with something else. We started to use "Config files API"
+
+Config files API (CFA) is a library written in ruby intended for accessing various configuration files. It is structured into several layers and creates an internal abstraction of configuration file. It has again been designed and developed in SUSE's YaST team. However this time it uses (or can use) third party parsers. CFA provides location transparency for the parser on the bottom layer and unific API for application on the top one.
+
+Lets go through layers in details.
+
+## First layer ##
+
+Is responsible for accessing configuration files itself. In simplest case it accesses local configuration files, but it can be adapted to access remote, chrooted or memory files too.
+
+## Second layer ##
+
+This layer parses the configuration file which was loaded by the underlying layer. It knows structure of the file and transforms it into abstract representation. The library typically uses external tool for parsing here (e.g. Augeas). So, if the external tools has a specific requirements, it has to be satisfied to get things working. For example if the Augeas is in use you need to provide him with a proper lens to parse the particular configuration file.
+
+## Third layer ##
+
+Last layer creates a model of the configuration file - an API for accessing configuration from an application. It basically creates "an abstraction on top of another abstraction". It means that it unifies use of various tools which can be used for accessing and parsing various configuration files.
+
+## Limitations ##
+
+The approach as described above has several limitations.
+
+Firstly you need to feed a parser too (e.g. provide lenses for augeas as described above). So, you at least has to know which parser is used for parsing which file and provide him with all stuff needed for parsing of file of your interest. This is especially important in case of nonstandard / custom configuration files.
```

Comment:
hm, I've mixed limitation for a developer and for a user ... I'll make it more obvious.